main.go

```diff
@@ -1,5 +1,58 @@
 package main
 
+import (
+	"context"
+	"log"
+	"os"
+
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/config"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/db"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/forman"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/reconciler"
+	"go.uber.org/zap"
+)
 
 func main() {
+	//logger := logr.Logger{}
+	ctx := context.Background()
+	configFilePath := os.Getenv("CONFIG_FILE_PATH")
+	if configFilePath == "" {
+		log.Fatal("CONFIG_FILE_PATH environment variable not set")
+	}
+	// parse configuration from yaml file
+	conf, err := config.LoadAndValidateConfig(configFilePath)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	zapConfig := zap.Config{
+		Encoding:         "json",
+		EncoderConfig:    zap.NewProductionEncoderConfig(),
+		OutputPaths:      []string{"stdout"},
+		ErrorOutputPaths: []string{"stderr"},
+	}
+	logger, err := zapConfig.Build()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	dbInstance, err := db.NewClient(ctx, &conf.DB)
+	if err != nil {
+		logger.Fatal(err.Error())
+	}
+	defer dbInstance.Close()
+
+	reqChan := forman.GoForman(logger, dbInstance, conf.Reconcile.Concurrency, reconciler.Reconcile)
```

Comment:
As mentioned in a previous comment. DB related structures should not be part of the Formen interface!

Reply:
i think there is some gap on this between us , I suggest having a meeting on what it looks like.

Reply:
No need for a meeting:
```
reqChan := forman.GoForman(logger, conf.Reconcile.Concurrency, func(req) {
    reconciler.reconcile(logger, dbInstance, req)
})
```